lucene/sandbox/src/java/org/apache/lucene/geo/Tessellator.java

```diff
@@ -248,8 +261,21 @@ private static final Node fetchHoleBridge(final Node holeNode, final Node outerN
         p = p.next;
       }
     }
+    //In case more than one hole uses this connection point,
+    // make sure we choose the right one
+    return getBestConnector(connection, holeNode);
+  }
 
-    return connection;
+  private static Node getBestConnector(Node connection, Node holeNode) {
+    Node candidate = connection;
+    Node next = connection.next;
+    do {
+      if (isVertexEquals(next, connection) && isLocallyInside(next, holeNode)) {
+        candidate = next;
+      }
+      next = next.next;
+    } while (next.next != connection);
+    return candidate;
```

Comment:
This can run in linear time with the number of vertices?

Reply:
Yes, it is 0(n). It does not seem to degrade the performance of my tests.